tests/hikari/impl/test_event_manager.py

```diff
@@ -227,6 +227,109 @@ async def test_on_channel_pins_update(self, stateless_event_manager, shard, even
+        self,
+        event_manager: event_manager_.EventManagerImpl,
+        shard: mock.Mock,
+        event_factory: mock.Mock,
+    ):
+        mock_payload = {"id": "123321"}
+        await event_manager.on_thread_create(shard, mock_payload)
+
+        event_manager.dispatch.assert_awaited_once_with(
+            event_factory.deserialize_guild_thread_access_event.return_value
+        )
+        event_factory.deserialize_guild_thread_access_event.assert_called_once_with(shard, mock_payload)
+
+    @pytest.mark.asyncio()
+    async def test_on_thread_update(
+        self,
+        event_manager: event_manager_.EventManagerImpl,
+        shard: mock.Mock,
+        event_factory: mock.Mock,
+    ):
+        mock_payload = mock.Mock()
+        await event_manager.on_thread_update(shard, mock_payload)
+
+        event_manager.dispatch.assert_awaited_once_with(
+            event_factory.deserialize_guild_thread_update_event.return_value
+        )
+        event_factory.deserialize_guild_thread_update_event.assert_called_once_with(shard, mock_payload)
+
+    @pytest.mark.asyncio()
+    async def test_on_thread_delete(
+        self,
+        event_manager: event_manager_.EventManagerImpl,
+        shard: mock.Mock,
+        event_factory: mock.Mock,
+    ):
+        mock_payload = mock.Mock()
+        await event_manager.on_thread_delete(shard, mock_payload)
+
+        event_manager.dispatch.assert_awaited_once_with(
+            event_factory.deserialize_guild_thread_delete_event.return_value
+        )
+        event_factory.deserialize_guild_thread_delete_event.assert_called_once_with(shard, mock_payload)
+
+    @pytest.mark.asyncio()
+    async def test_on_thread_list_sync(
+        self,
+        event_manager: event_manager_.EventManagerImpl,
+        shard: mock.Mock,
+        event_factory: mock.Mock,
+    ):
+        mock_payload = mock.Mock()
+        await event_manager.on_thread_list_sync(shard, mock_payload)
+
+        event_manager.dispatch.assert_awaited_once_with(event_factory.deserialize_thread_list_sync_event.return_value)
+        event_factory.deserialize_thread_list_sync_event.assert_called_once_with(shard, mock_payload)
+
+    @pytest.mark.asyncio()
+    async def test_on_thread_member_update(
+        self,
+        event_manager: event_manager_.EventManagerImpl,
+        shard: mock.Mock,
+        event_factory: mock.Mock,
+    ):
+        mock_payload = mock.Mock()
+        await event_manager.on_thread_member_update(shard, mock_payload)
+
+        event_manager.dispatch.assert_awaited_once_with(
+            event_factory.deserialize_own_thread_member_update_event.return_value
+        )
+        event_factory.deserialize_own_thread_member_update_event.assert_called_once_with(shard, mock_payload)
+
+    @pytest.mark.asyncio()
+    async def test_on_thread_members_update(
+        self,
+        event_manager: event_manager_.EventManagerImpl,
+        shard: mock.Mock,
+        event_factory: mock.Mock,
```

Comment:
I completely understand that they should be typed, but it looks weird to have 90% of the tests untyped and these typed. If we do it, I would rather type all of them and not only do some

Reply:
TLDR i think it'd make more sense to just gradually add type hints to the tests over time rather than just like bulk do it once someone has enough time, spare sanity and motivation to update all theses tests + tests that'll be added in the future at once

Reply:
Completely understandable, and I agree. Just think it looks weird. Will try to do some program to get most added. For now keep them and we shall see how it evolves